docs/book/src/plugins/external-plugins.md

```diff
@@ -0,0 +1,155 @@
+
+- If you are looking for implementing plugins by a different language other than `Go`.
+
+## How to write it?
+
+The inter-process communication between your external plugin and Kubebuilder is through the standard I/O.
+
+You can write your external plugin in any language as long as it implements the `PluginRequest` and `PluginResponse` interfaces. 
+
+`PluginRequest` holds all the information Kubebuilder receives from the CLI and the plugins that were executed before it.
+The marshaled `PluginRequest` (which should be a `JSON` object) would be sent over `stdin` to the external plugin by Kubebuilder.
+
+```go
+// PluginRequest contains all information kubebuilder received from the CLI
+// and plugins executed before it.
+type PluginRequest struct {
+	// APIVersion defines the versioned schema of PluginRequest that is being sent from Kubebuilder.
+	// Initially, this will be marked as alpha (v1alpha1).
+	APIVersion string `json:"apiVersion"`
+
+	// Args holds the plugin specific arguments that are received from the CLI
+	// which are to be passed down to the external plugin.
+	Args []string `json:"args"`
+
+	// Command contains the command to be executed by the plugin such as init, create api, etc.
+	Command string `json:"command"`
+
+	// Universe represents the modified file contents that gets updated over a series of plugin runs
+	// across the plugin chain. Initially, it starts out as empty.
+	Universe map[string]string `json:"universe"`
+}
+```
+
+`PluginResponse` is what the plugin constructs with the updated universe and serialized and sent back to Kubebuilder through `stdout`. 
+
+```go
+// PluginResponse is returned to kubebuilder by the plugin and contains all files
+// written by the plugin following a certain command.
+type PluginResponse struct {
+	// APIVersion defines the versioned schema of the PluginResponse that is back sent back to Kubebuilder.
+	// Initially, this will be marked as alpha (v1alpha1)
+	APIVersion string `json:"apiVersion"`
+
+	// Command holds the command that gets executed by the plugin such as init, create api, etc.
+	Command string `json:"command"`
+
+	// Metadata contains the plugin specific help text that the plugin returns to Kubebuilder when it receives
+	// `--help` flag from Kubebuilder.
+	Metadata plugin.SubcommandMetadata `json:"metadata"`
+
+	// Universe in the PluginResponse represents the updated file contents that was written by the plugin.
+	Universe map[string]string `json:"universe"`
+
+	// Error is a boolean type that indicates whether there were any errors due to plugin failures.
+	Error bool `json:"error,omitempty"`
+
+	// ErrorMsgs contains the specific error messages of the plugin failures.
+	ErrorMsgs []string `json:"errorMsgs,omitempty"`
+
+	// Flags contains the plugin specific flags that the plugin returns to Kubebuilder when it receives
+	// a request for a list of supported flags from Kubebuilder
+	Flags []Flag `json:"flags,omitempty"`
+}
+```
```

Comment:
It might be nice to include an example of the JSON for both of these. It might be difficult for someone reading the docs to visualize the JSON based on Go type definitions.

Reply:
Good idea!
I added sample JSONs for both request & response together with explainations.
Also, the Go definitions are now referred in the link instead of directly pasting them in the docs.

Hope this can make it clearer..